### PR TITLE
apimachinery: move unversioned registration to metav1

### PR DESCRIPTION
**Pull request by sttts**
Follow-up from the discussions in https://github.com/kubernetes/kubernetes/pull/43027:

We need `Status` as unversioned type which is hardcoded to `GroupVersion{Group: "", Version: "v1"}`. If the core group is not in the scheme, we miss `Status`.

Fixing https://github.com/kubernetes/kubernetes/issues/47030.

### CODE REVIEW

**Comment by seh** [comment]
[Related discussion](https://kubernetes.slack.com/archives/C0EG7JC6T/p1495206979665958) from the "sig-api-machinery" channel in the "Kubernetes" Slack team.

**Comment by smarterclayton** [comment]
Scary, because this may alter backwards compat if we get it even the slightest bit wrong.  Code looks correct though.

**Comment by sttts** [comment]
> Scary, because this may alter backwards compat

I don't propose behaviour change (yet), only to move the code and register it with the other meta types.

**Comment by smarterclayton** [comment]
I just mean, verify that the behavior of the system hasn't changed by this
move (in how status is returned).  It looks right - but this affects
default behavior of old clients potentially.

**Comment by seh** [comment]
Were any of those test failures due to the changes proposed here?

I'm wondering what the next step will be to get this merged.

**Comment by sttts** [comment]
@seh seeing only `Can't connect to 'docker' daemon.  please fix and retry.`.

**Comment by sttts** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by sttts** [comment]
@smarterclayton added an integration test that the Status objects look like they should. Anything else you want to have tested?

**Comment by deads2k** [comment]
@smarterclayton are you blocking this for 1.7?

**Comment by smarterclayton** [comment]
Not blocking, just no time to review

/approve

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/register.go [hunk 1]
```diff
@@ -56,6 +60,15 @@ func AddToGroupVersion(scheme *runtime.Scheme, groupVersion schema.GroupVersion)
 		Convert_versioned_Event_to_versioned_InternalEvent,
 	)
 
+	// Register Unversioned types under their own special group
+	scheme.AddUnversionedTypes(Unversioned,
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/register.go]: And this is safe to call multiple times for the same scheme?

File: staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go [hunk 2]
```diff
@@ -151,8 +151,8 @@ func (s *Scheme) AddUnversionedTypes(version schema.GroupVersion, types ...Objec
 		t := reflect.TypeOf(obj).Elem()
 		gvk := version.WithKind(t.Name())
 		s.unversionedTypes[t] = gvk
-		if _, ok := s.unversionedKinds[gvk.Kind]; ok {
-			panic(fmt.Sprintf("%v has already been registered as unversioned kind %q - kind name must be unique", reflect.TypeOf(t), gvk.Kind))
+		if old, ok := s.unversionedKinds[gvk.Kind]; ok && t != old {
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go]: ah, I see.  You made it safe.

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by nikhita** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by nikhita** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**